yt/__init__.py

```diff
@@ -9,10 +9,9 @@
 """
 import sys
 
-if sys.version_info[0] < 3:
-    raise Exception(
-        "Python 2 no longer supported.  Please install Python 3 for use with yt."
-    )
+py_version = sys.version_info
+if not (py_version.major == 3 and py_version.minor >= 7):
```

Comment:
Maybe we should make this non-fatal and just say, caveat emptor

Reply:
I was actually thinking the opposite, since the current `raise Exception` is catchable, so I replaced it with `sys.exit` specifically to make it fatal. I figure that when we _do_ switch to 3.7 as minimal, the code base will quickly become unstable against with 3.6 and `import yt` will fail no matter what, so I do think it's preferable to exit early with a controlled error message.

Reply:
Any reason not to use 
```suggestion
if py_version < (3, 7):
```
I _know_ you're going to mention Python 4, but to quote Guido (https://twitter.com/gvanrossum/status/1306270482384076800?s=20)
>  If there ever is a version 4, the transition from 3 to 4 will be more like that from 1 to 2 rather than 2 to 3.

Reply:
Yup I'm aware Python 4.0 is not proposed yet, but I still think it's better to assume the worst case scenario.

Reply:
I'm fine with your solution though ;)